ironfish/src/rpc/routes/wallet/exportAccount.ts

```diff
@@ -43,31 +48,29 @@ routes.register<typeof ExportAccountRequestSchema, ExportAccountResponse>(
     AssertHasRpcContext(request, node, 'wallet')
 
     const account = getAccount(node.wallet, request.data.account)
-    const { id: _, ...accountInfo } = account.serialize()
-    if (request.data.viewOnly) {
-      accountInfo.spendingKey = null
-      if (accountInfo.multisigKeys) {
-        accountInfo.multisigKeys = {
-          publicKeyPackage: accountInfo.multisigKeys.publicKeyPackage,
-        }
-      }
-    }
 
-    if (!request.data.format) {
-      let createdAt = null
-      if (accountInfo.createdAt) {
-        createdAt = {
-          hash: accountInfo.createdAt.hash.toString('hex'),
-          sequence: accountInfo.createdAt.sequence,
-        }
-      }
+    if (request.data.format) {
+      const value = toAccountImport(account)
 
-      request.end({ account: { ...accountInfo, createdAt } })
-    } else {
-      const encoded = encodeAccount(accountInfo, request.data.format, {
+      const encoded = encodeAccountImport(value, request.data.format, {
+        viewOnly: request.data.viewOnly,
         language: request.data.language,
       })
+
       request.end({ account: encoded })
+      return
     }
+
+    // Backwards compatability path we must send back an RpcAccountImport
+
+    const serialized = serializeRpcImportAccount(
+      decodeAccountImport(
+        encodeAccountImport(toAccountImport(account), AccountFormat.JSON, {
```

Comment:
Does this need to encode/decode the account import? Looks like this would allow for adding support for viewOnly, but the old codepath didn't have that. I'm okay either way, just noticed the nested functions

Reply:
I think the old code path supported viewOnly here. https://github.com/iron-fish/ironfish/blob/8b26bfead96c577399e4cccd5879dcb785194307/ironfish/src/rpc/routes/wallet/exportAccount.ts#L47-L54

The problem I'm trying to solve is to remove all the duplicated code for how to encode twice. If I move support for viewOnly to `toAccountImport` it would remove duplication. I was thinking of doing that in the next PR to avoid this.

Reply:
Fixed by moving this into toAccountImport